designer/client/src/components/MenuBar/TabElement.tsx

```diff
@@ -1,21 +1,32 @@
 import { styled } from "@mui/material";
 import React from "react";
-import { NavLink } from "react-router-dom";
+import { NavLink, useLocation } from "react-router-dom";
 
 import type { DynamicTabData } from "../../containers/DynamicTab";
 
 function UnstyledTabElement({ tab, ...props }: { tab: DynamicTabData; className?: string }): JSX.Element {
-    const { id, type, url, title } = tab;
+    const { id, type, url, title, currentLocationInQuery } = tab;
```

Comment:
Maybe we can move it outside of the React component since it's not related to the react state or props?

```
export function getTabNavigationUrl(tab: DynamicTabData): string {
    const { url, currentLocationInQuery } = tab;

    const fullOriginalPath = window.location.pathname + window.location.search + window.location.hash;
    const fullOriginalUrl = window.location.origin + fullOriginalPath;

    if (!currentLocationInQuery || !currentLocationInQuery.enabled) {
        return url;
    }

    const enrichedUrl = new URL(url);
    enrichedUrl.searchParams.set(currentLocationInQuery.parameterName, fullOriginalUrl);
    return enrichedUrl.toString();
}
```

const navigationUrl = getTabNavigationUrl(tab);